### DISCUSSION TITLE
Strange behavior in circuit with Pull-Down Resistor

### DISCUSSION THREAD

**Discussion by RicardoKers** [hneemann/Digital#1041] [General]
I'm finding strange behavior in circuit with Pull-Down Resistor. It is a decoder circuit for a 16-key matrix keyboard. The circuit is simple:

![image](https://user-images.githubusercontent.com/49044489/194872825-f9c41aa9-276c-4f28-966c-12d473052ef6.png)

The objective is to show the number of the pressed key on the display.
The circuit works correctly for lines 2, 3 and 4 (switches 4 to F), but when I activate a key of the first line (switches 0 to 3) the circuit presents a strange behavior, storing in the display the number of the selected key in the first line and also the corresponding key number in the third line. For example, when I select switch 1, the display receives 1 and 9 alternately.
I don't know what I'm doing wrong.

[LeTecMat.zip](https://github.com/hneemann/Digital/files/9746208/LeTecMat.zip)

**Comment by mengstr** [comment]
Your 2 bit row counter is not synchronous so the two bits from it dosen't change at the absolute same time, but does that with a little bit of delay between the two bits due to the propagation time in the gates.

If you hook up Probes in Counter mode to your four rows you'll see that they are getting extra pulses caused by the four AND gates triggering on the false conditions while the counter bits are not in sync for a short while.

<img width="673" alt="Screenshot 2022-10-10 at 16 14 59" src="https://user-images.githubusercontent.com/325326/194887735-3b580793-1233-4ff0-902e-257f7ac4a76e.png">

**Reply by RicardoKers** [reply]
Really, I hadn't remembered that. I teach about the problems of asynchronous counters to my students and I didn't apply it in this circuit, what a shame. 
I switched to a synchronous counter and now it works perfectly. 
Thanks.
